wdns/ede_purpose.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2023 DomainTools LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef WDNS_EDE_purpose_H
+#define WDNS_EDE_purpose_H
+
+typedef char * ede_purpose;
```

Comment:
This `typedef` adds unnecessary indirection and also changes the type of `ede_purpose_array[]`.

You want `const char * ede_purpose_array[]` so that `ede_purpose_array[i]` is a `const char *`. The `char *` typedef makes `ede_purpose_array` a constant array of non-const `char *`.

Reply:
How about changing the typedef to be `typedef const char * ede_purpose;`?

Which should make `ede_purpose_array` a constant array of constant char *. This was the type I was originally going for since the array itself shouldn't be changed at runtime.

Reply:
That would fix the element type, but `const char *ede_purpose[] = { ... }` is clearer and more idiomatic for an array of string literals. Making the array itself constant is an intriguing goal, and if that's the goal of the typdef it should be noted in a comment.  That said, the array is  _mostly_ protected from modification by its limited scope (if rolled in to `edns_options.c`) and read-only use, and I think that's adequate in this case.

(In contrast, the `const char *` is still valuable as those strings are passed to functions for which it is more difficult to verify read-only use, unless they take `const char *` parameters)

Reply:
Ok; I pushed a commit to change this.